### DISCUSSION TITLE
Do the SENSOR or SEARCH nodes need Internet access in non-airgap scenario (at all?)

### DISCUSSION THREAD

**Discussion by AnyrunMalware** [Security-Onion-Solutions/securityonion#5926] [Unsupported Versions] [answered]
I am planning on having a distributed grid deployment (Manager, Sensor, Search).

My goal is to only have the management node have internet connectivity and have all nodes go through the manager for updates (all updates, not just the yum updates).

Is that possible? OR do they still require internet access for suricate updates for example?

During the installation, we are prompted to answer this question:
.
**How would you like to connect to the Internet ?
.
"Direct" - Internet requests connect directly to the internet.
"Direct + Manager" - All traffic passes to the internet normally, but yum updates will instead be pulled from manager.
"Proxy" - proxy the traffic for git, docker client, wget, curl, yum, and various other SO components through a separate server in your environment.
"Proxy + Manager" - proxy all traffic from the "Proxy" option except yum updates, which will instead pull from the manager.**
.
.
Here is my confusion, the install document says this:

Internet Communication
.
When configuring network firewalls for Internet-connected deployments (non-airgap), you’ll want to ensure that nodes can connect outbound to the following:
.
repo.securityonion.net (CentOS Updates)
raw.githubusercontent.com (Security Onion public key)
sigs.securityonion.net (Signature files for Security Onion containers)
ghcr.io (Container downloads)
rules.emergingthreatspro.com (Emerging Threats IDS rules)
www.snort.org (Paid Snort Talos ruleset)
github.com (Strelka and Sigma rules updates)
notary.kolide.co (osquery agent update)
Ubuntu PPAs (OS Updates - Ubuntu only)
download.docker.com (Docker packages - Ubuntu only)
repo.saltstack.com (Salt packages - Ubuntu only)
packages.wazuh.com (Wazuh packages - Ubuntu only)

...

Thoughts?

Any  help is appreciated. Thanks!

**Comment by dougburks** [comment, marked as answer]
From https://github.com/Security-Onion-Solutions/securityonion/discussions/5876#discussioncomment-1494850:

I think the "Direct + Manager" option will work for your use case.

https://docs.securityonion.net/en/2.3/firewall.html#internet-communication is really referring to the deployment as a whole. So if your manager has Internet access and you specify "Direct + Manager" for your other nodes then they should be able to get everything they need from the manager.
